core/src/main/scripts/import_data_validator/validateData.py

```diff
@@ -1220,24 +1231,165 @@ class ClinicalValidator(Validator):
+                    if value not in VALID_ATTR_TYPES:
+                        self.logger.error(
+                            'Invalid attribute type definition, must be one of'
+                            ' [%s]',
+                            ', '.join(VALID_ATTR_TYPES),
+                            extra={'line_number': line_index + 1,
+                                   'colum_number': col_index + 1,
+                                   'cause': value})
+                        invalid_values = True
+                elif LINE_NAMES[line_index] == 'priority':
+                    try:
+                        if int(value) < 1:
+                            raise ValueError()
+                    except ValueError:
+                        self.logger.error(
+                            'Priority definition must be a positive integer',
+                            extra={'line_number': line_index + 1,
+                                   'column_number': col_index + 1,
+                                   'cause': value})
+                        invalid_values = True
+                else:
+                    raise ValueError('Unknown clinical header line name')
+
+                attr_defs[col_index][LINE_NAMES[line_index]] = value
+
+        self.attr_defs = attr_defs
+        return not invalid_values
 
     def checkHeader(self, cols):
+
         num_errors = super(ClinicalValidator, self).checkHeader(cols)
-        for col_name in self.cols:
+
+        if self.numCols != len(self.attr_defs):
+            self.logger.error(
+                'Varying numbers of columns in clinical header (%d, %d)',
+                len(self.attr_defs),
+                len(self.cols),
+                extra={'line_number': self.line_number})
+            num_errors += 1
+        # use the non-file-related base logger for this class method
+        self.request_attrs(self.logger.logger)
+        for col_index, col_name in enumerate(self.cols):
             if not col_name.isupper():
                 self.logger.warning(
                     "Clinical header not in all caps",
                     extra={'line_number': self.line_number,
                            'cause': col_name})
-        self.cols = [s.upper() for s in self.cols]
+            srv_attr = self.srv_attrs.get(col_name)
+            if srv_attr is None:
+                self.logger.warning(
+                    'New %s-level attribute will be added to the portal',
+                    self.attr_defs[col_index]['attribute_type'].lower(),
+                    extra={'line_number': self.line_number,
+                           'column_number': col_index + 1,
+                           'cause': col_name})
+            else:
+                # copy the dict to make local changes
+                srv_attr = dict(srv_attr)
+                # compare defined values with existing ones
+                if 'attribute_type' in srv_attr:
+                    raise ValueError(
```

Comment:
Will do, my comment there is indeed confusing. The reason is that this would not be an error in the staging files, but a change in the response of the server API that the validator script will have to be adapted to.